gpflux/covariances/multioutput/cvvs.py

```diff
@@ -0,0 +1,185 @@
+                kernel.kernel,
+                L_Kuu=l_kuu,
+            )
+            for k, l_kuu in zip(kernel.kernels, L_Kuu)
+        ],
+        axis=0,
+    )
+
+    jittermat = tf.eye(inducing_variable_v.num_inducing, dtype=Cvv.dtype)[None, :, :] * jitter
+    return Cvv + jittermat
+
+
+@Cvv.register(
+    FallbackSeparateIndependentInducingVariables,
+    FallbackSeparateIndependentInducingVariables,
+    SharedIndependent,
+)
+@check_shapes(
+    "inducing_variable_u: [M_u, D, P]",
+    "inducing_variable_v: [M_v, D, P]",
+    "return: [L, M_v, M_v]",
+)
+def Kuu_fallback_separate_shared(
+    inducing_variable_u: FallbackSeparateIndependentInducingVariables,
+    inducing_variable_v: FallbackSeparateIndependentInducingVariables,
+    kernel: SharedIndependent,
+    *,
+    jitter: float = 0.0,
+    L_Kuu: Optional[tf.Tensor] = None,
+) -> tf.Tensor:
+
+    Cvv = tf.stack(
+        [
+            Cvv(ind_var_u, ind_var_v, kernel.kernel, L_Kuu=l_kuu)
+            for ind_var_u, ind_var_v, l_kuu in zip(
+                inducing_variable_u.inducing_variable_list,
+                inducing_variable_v.inducing_variable_list,
+                L_Kuu,
+            )
+        ],
+        axis=0,
+    )
+
+    jittermat = tf.eye(inducing_variable_v.num_inducing, dtype=Cvv.dtype)[None, :, :] * jitter
+    return Cvv + jittermat
+
+
+@Cvv.register(
+    FallbackSeparateIndependentInducingVariables,
+    FallbackSeparateIndependentInducingVariables,
+    SeparateIndependent,
+)
+@check_shapes(
+    "inducing_variable_u: [M_u, D, P]",
+    "inducing_variable_v: [M_v, D, P]",
+    "return: [L, M_v, M_v]",
+)
+def Kuu_fallback_separate(
+    inducing_variable_u: FallbackSeparateIndependentInducingVariables,
+    inducing_variable_v: FallbackSeparateIndependentInducingVariables,
+    kernel: SeparateIndependent,
+    *,
+    jitter: float = 0.0,
+    L_Kuu: Optional[tf.Tensor] = None,
+) -> tf.Tensor:
+    n_iv_u = len(inducing_variable_u.inducing_variable_list)
+    n_iv_v = len(inducing_variable_v.inducing_variable_list)
+    n_k = len(kernel.kernels)
+    assert (
+        n_iv_u == n_k
+    ), f"Must have same number of inducing variables and kernels. Found {n_iv_u} and {n_k}."
+
+    assert (
+        n_iv_v == n_k
+    ), f"Must have same number of inducing variables and kernels. Found {n_iv_v} and {n_k}."
+
+    Cvv = tf.stack(
+        [
+            Cvv(ind_var_u, ind_var_v, k, L_Kuu=l_kuu)
+            for ind_var_u, ind_var_v, l_kuu, k in zip(
+                inducing_variable_u.inducing_variable_list,
+                inducing_variable_v.inducing_variable_list,
+                L_Kuu,
+                kernel.kernels,
+            )
+        ],
+        axis=0,
+    )
+
+    jittermat = tf.eye(inducing_variable_v.num_inducing, dtype=Cvv.dtype)[None, :, :] * jitter
+    return Cvv + jittermat
+
+    Kmms = [Kuu(f, k) for f, k in zip(inducing_variable.inducing_variable_list, kernel.kernels)]
```

Comment:
@SebastianPopescu is this `inducing_variable_u` or `inducing_variable_v`?

Reply:
uups..sorry..I again forgot to remove line 182